### ISSUE TITLE
Support Yarn PnP

### ISSUE THREAD

**Issue by ezracelli** [telefunc/telefunc#77]
PnP (a module installation strategy used by Yarn Berry) stores modules in `.zip` files. When a package is imported, the package manager creates an isolated, read-only virtual filesystem. More info: https://yarnpkg.com/features/pnp

`telefunc/vite#plugin` has a side-effect that attempts to write to one of `telefunc`'s own files. Due to the read-only nature of the virtual filesystem created by PnP, this write fails.

https://github.com/brillout/telefunc/blob/132764550bd18286f3fa321641b66145ee0d4ed6/telefunc/node/vite/index.ts#L20-L21

https://github.com/brillout/telefunc/blob/132764550bd18286f3fa321641b66145ee0d4ed6/telefunc/node/vite/importGlob/toggle.ts#L21-L22

### Workaround

<details>
<summary>
I patched the module manually.
</summary>

```patch
diff --git a/dist/cjs/node/vite/importGlob/telefuncFilesGlob.js b/dist/cjs/node/vite/importGlob/telefuncFilesGlob.js
index 12d2beb0902533d560a37205d5c1779a41192d46..a9369b2a79929fd8dc53f02a50bc5d6cf000ae84 100644
--- a/dist/cjs/node/vite/importGlob/telefuncFilesGlob.js
+++ b/dist/cjs/node/vite/importGlob/telefuncFilesGlob.js
@@ -1,4 +1 @@
-"use strict";
-Object.defineProperty(exports, "__esModule", { value: true });
-exports.importGlobUnset = void 0;
-exports.importGlobUnset = true;
+export const telefuncFilesGlob = import.meta.glob("/**/*.telefunc.(js|cjs|mjs|ts|cts|mts|jsx|cjsx|mjsx|tsx|ctsx|mtsx)");
diff --git a/dist/cjs/node/vite/index.js b/dist/cjs/node/vite/index.js
index dd988d6ee4dd2dd361d5461df11d9742ab18b940..83523a7ccfadf5486f2899699dcd4ac3ba55c1e3 100644
--- a/dist/cjs/node/vite/index.js
+++ b/dist/cjs/node/vite/index.js
@@ -12,10 +12,8 @@ const importBuild_1 = require("./plugins/importBuild");
 const previewConfig_1 = require("./plugins/previewConfig");
 const printShieldGenResult_1 = require("./plugins/printShieldGenResult");
 const manifest_1 = require("./plugins/manifest");
-const toggle_1 = require("./importGlob/toggle");
 // Return as `any` to avoid Plugin type mismatches when there are multiple Vite versions installed
 function plugin(config) {
-    (0, toggle_1.importGlobOn)();
     const plugins = [
         (0, transform_1.transform)(),
         (0, commonConfig_1.commonConfig)(),
```
</details>

**Comment by brillout** [comment]
Indeed. FYI the side-effect will be removed soon. In the meantime use pnpm (recommended) or npm instead.

**Comment by rtritto** [comment]
@ezracelli @dvteixeira24 is this fix also compatible with npm and pnpm?

**Comment by rtritto** [comment]
> Indeed. FYI the side-effect will be removed soon. In the meantime use pnpm (recommended) or npm instead.

@brillout also pnpm can use PnP with `node-linker=pnp` (https://pnpm.io/npmrc#node-linker).

Can I help with a PR to remove the side-effects? If yes, there are some guidelines?

**Comment by brillout** [comment]
> will be removed soon

I got sidetracked by Vike's success. It's still very much on the radar, but it requires some complex restructuring (breaking down the Telefunc package into multiple packages `@telefunc/vite`, `@telefunc/next`, ....). I'll work on it after Vike hits `1.0` (ETA Q1).

> Can I help with a PR to remove the side-effects?

That'd be great. Although, for this one, it's better I do it. Would you be up to contribute on something else?

**Comment by brillout** [comment]
> Can I help with a PR to remove the side-effects?

Actually, we can give it a try. Let's see how it goes 👀 But let's be open to finish the PR later if it get's too complex.

> If yes, there are some guidelines?

Yes, have a look at Telefunc's Vite plugin, you'll see the code that writes to `node_modules/`. You can try to remove that code while making the CI happy. Let's see how far we can get without major restructuring.

> Would you be up to contribute on something else?

See https://github.com/vikejs/vike/issues/1349, most notably:
 - [New Vike extensions](https://github.com/vikejs/vike/issues/1715)
 - [New Bati integrations](https://github.com/vikejs/bati/issues/456)
 - [Eject](https://github.com/snake-py/eject/issues/4) (this is actually fairly high priority and I can see you liking this 👀)

Looking forward to it!

**Comment by rtritto** [comment]
> Actually, we can give it a try. Let's see how it goes 👀 But let's be open to finish the PR later if it get's too complex.

Since Telefunc will be splitted in multiple packages, it doesn't seem to make sense to start a PR now.
Actually, for this issue, I can't use Telefunc in my projects (all PnP based), but I can wait and work on other things.

---

Thanks for the recap of the activities, I will take a look at where I can help.

**Comment by brillout** [comment]
It's already done, Telefunc now uses a virtual module instead of writing to the FS.
